kubernetes/structures.go

```diff
@@ -12,6 +12,8 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+var permittedInternalAnnotations = map[string]bool{}
```

Comment:
Please, please no globals!

Also, this doesn't need to be a map. It's a whitelist, so including a value in it implies the `true`. It can just be a slice of strings.

An alternative would be to embed the `kubernetes.Clientset` type returned by `kubernetes.NewForConfig()` in a new type and include the `PermittedInternalAnnotations` in it as well. That way the meta value passed around the provider still behaves like a Clientset and you can also carry this list around.

You could define a type like:
```go
type metaClientset struct {
	kubernetes.Clientset
	PermittedInternalAnnotations []string
}
```
and then do something like:
```go
	k, err := kubernetes.NewForConfig(cfg)
	if err != nil {
		return nil, fmt.Errorf("Failed to configure: %s", err)
	}
	mc := metaClientset{
		Clientset: *k,
	}
	if v, ok := d.GetOk("internal_annotations_whitelist"); ok {
		vs := v.(*schema.Set).List()
		for _, raw := range vs {
			mc.PermittedInternalAnnotations = append(mc.PermittedInternalAnnotations, raw.(string))
		}
	}
	return &mc, nil
```
at the end of `providerConfigure()`

Reply:
The global was used because the metadata structure is not available in the validation code path.  With the current validation approach, `terraform plan` generates an error if the annotation fails to pass the IsQualifiedName() test.  I wasn't able to figure out how to accomplish this without the use of a validator function.

True, a slice of strings is fine provided you don't want to enforce uniqueness in the list (and you don't want to add blacklist functionality using the same list).